<!DOCTYPE html>
<html lang="en">
  

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width,minimum-scale=1">

  <title>dbGaP-download</title>
  <meta name="description" content="">

  <link rel="canonical" href="http://0.0.0.0:4000/lab-wiki/project_resource/dbGaP-download.html">
  <link rel="alternate" type="application/rss+xml" title="Lab Docs" href="http://0.0.0.0:4000/lab-wiki/feed.xml">

  <meta property="og:url"         content="http://0.0.0.0:4000/lab-wiki/project_resource/dbGaP-download.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="dbGaP-download" />
<meta property="og:description" content="" />
<meta property="og:image"       content="" />


  <script type="application/ld+json">
  {
  "@context": "http://schema.org",
  "@type": "NewsArticle",
  "mainEntityOfPage":
    "http://0.0.0.0:4000/lab-wiki/project_resource/dbGaP-download.html",
  "headline":
    "dbGaP-download",
  "datePublished":
    "2019-12-17T11:05:49-06:00",
  "dateModified":
    "2019-12-17T11:05:49-06:00",
  "description":
    "",
  "author": {
    "@type": "Person",
    "name": "Gao Wang"
  },
  "publisher": {
    "@type": "Organization",
    "name": "Data 100 at UC Berkeley",
    "logo": {
      "@type": "ImageObject",
      "url": "http://0.0.0.0:4000/lab-wiki",
      "width": 60,
      "height": 60
    }
  },
  "image": {
    "@type": "ImageObject",
    "url": "http://0.0.0.0:4000/lab-wiki",
    "height": 60,
    "width": 60
  }
}

  </script>
  <link rel="stylesheet" href="/lab-wiki/assets/css/styles.css">
  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css ">
  <link rel="apple-touch-icon" sizes="57x57" href="/apple-touch-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/apple-touch-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/apple-touch-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/apple-touch-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/apple-touch-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/apple-touch-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/apple-touch-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/apple-touch-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon-180x180.png">

  <!-- <link rel="manifest" href="/manifest.json"> -->
  <!-- <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#efae0a"> -->
  <meta name="msapplication-TileColor" content="#da532c">
  <meta name="msapplication-TileImage" content="/mstile-144x144.png">
  <meta name="theme-color" content="#233947">

  <!-- Favicon -->
  <link rel="shortcut icon" type="image/x-icon" href="/lab-wiki/images/favicon.ico">

  <!-- MathJax Config -->
  <!-- Allow inline math using $ and automatically break long math lines -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
    tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true,
        processEnvironments: true
    },
    CommonHTML: {
        linebreaks: {
            automatic: true,
        },
    },
});
</script>
<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS_CHTML' async></script>

  <!-- DOM updating function -->
  <script>
const runWhenDOMLoaded = cb => {
  if (document.readyState != 'loading') {
    cb()
  } else if (document.addEventListener) {
    document.addEventListener('DOMContentLoaded', cb)
  } else {
    document.attachEvent('onreadystatechange', function() {
      if (document.readyState == 'complete') cb()
    })
  }
}

// Helper function to init things quickly
initFunction = function(myfunc) {
  runWhenDOMLoaded(myfunc);
  document.addEventListener('turbolinks:load', myfunc);
};
</script>

  <!-- Define some javascript variables that will be useful in other javascript -->
  <script>
    const site_basename = '/lab-wiki';
  </script>

  <!-- Add AnchorJS to let headers be linked -->
  <script src="/lab-wiki/assets/js/anchor.min.js"  type="text/javascript"></script>
  <script>

initFunction(function () {
    anchors.add("main h1, main h2, main h3, main h4")
});

</script>

  <!-- Include Turbolinks to make page loads fast -->
  <!-- https://github.com/turbolinks/turbolinks -->
  <script src="/lab-wiki/assets/js/turbolinks.js" async></script>
  <meta name="turbolinks-cache-control" content="no-cache">

  <!-- Load nbinteract for widgets -->
  

  <!-- Load Thebelab for interactive widgets -->
  <!-- Include Thebelab for interactive code if it's enabled -->



  <!-- Load the auto-generating TOC -->
  <script src="/lab-wiki/assets/js/tocbot.min.js"  type="text/javascript"></script>
  <script>
var initToc = function () {
  tocbot.init({
    tocSelector: 'nav.onthispage',
    contentSelector: '.c-textbook__content',
    headingSelector: 'h2, h3',
    orderedList: false,
    collapseDepth: 6,
    listClass: 'toc__menu',
    activeListItemClass: "",  // Not using
    activeLinkClass: "", // Not using
  });
  tocbot.refresh();
}
initFunction(initToc);
</script>

  <!-- Google analytics -->
  <script src="/lab-wiki/assets/js/ga.js" async></script>

  <!-- Clipboard copy button -->
  <script src="https://cdn.jsdelivr.net/npm/clipboard@2/dist/clipboard.min.js" async></script>

  <!-- Load JS that depends on site variables -->
  <script>
/**
 * Set up copy/paste for code blocks
 */
const codeCellId = index => `codecell${index}`

const clipboardButton = id =>
  `<a id="copy-button-${id}" class="btn copybtn o-tooltip--left" data-tooltip="Copy" data-clipboard-target="#${id}">
    <img src="/lab-wiki/assets/images/copy-button.svg" alt="Copy to clipboard">
  </a>`

// Clears selected text since ClipboardJS will select the text when copying
const clearSelection = () => {
  if (window.getSelection) {
    window.getSelection().removeAllRanges()
  } else if (document.selection) {
    document.selection.empty()
  }
}

// Changes tooltip text for two seconds, then changes it back
const temporarilyChangeTooltip = (el, newText) => {
  const oldText = el.getAttribute('data-tooltip')
  el.setAttribute('data-tooltip', newText)
  setTimeout(() => el.setAttribute('data-tooltip', oldText), 2000)
}

const addCopyButtonToCodeCells = () => {
  // If ClipboardJS hasn't loaded, wait a bit and try again. This
  // happens because we load ClipboardJS asynchronously.
  if (window.ClipboardJS === undefined) {
    setTimeout(addCopyButtonToCodeCells, 250)
    return
  }

  const codeCells = document.querySelectorAll('div.c-textbook__content > div.highlighter-rouge > div.highlight > pre, div.input_area pre')
  codeCells.forEach((codeCell, index) => {
    const id = codeCellId(index)
    codeCell.setAttribute('id', id)
    if (document.getElementById("copy-button" + id) == null) {
      codeCell.insertAdjacentHTML('afterend', clipboardButton(id));
    }
  })

  const clipboard = new ClipboardJS('.copybtn')
  clipboard.on('success', event => {
    clearSelection()
    temporarilyChangeTooltip(event.trigger, 'Copied!')
  })

  clipboard.on('error', event => {
    temporarilyChangeTooltip(event.trigger, 'Failed to copy')
  })

  // Get rid of clipboard before the next page visit to avoid memory leak
  document.addEventListener('turbolinks:before-visit', () =>
    clipboard.destroy()
  )
}

initFunction(addCopyButtonToCodeCells);
</script>


  <!-- Hide cell code -->
  <script>
    /**
    Add buttons to hide code cells
    */


    var setCodeCellVisibility = function (inputField, kind) {
        // Update the image and class for hidden
        var id = inputField.getAttribute('data-id');
        var codeCell = document.querySelector(`#${id} div.highlight`);

        if (kind === "visible") {
            codeCell.classList.remove('hidden');
            inputField.checked = true;
        } else {
            codeCell.classList.add('hidden');
            inputField.checked = false;
        }
    }

    var toggleCodeCellVisibility = function (event) {
        // The label is clicked, and now we decide what to do based on the input field's clicked status
        if (event.target.tagName === "LABEL") {
            var inputField = event.target.previousElementSibling;
        } else {
            // It is the span inside the target
            var inputField = event.target.parentElement.previousElementSibling;
        }

        if (inputField.checked === true) {
            setCodeCellVisibility(inputField, "visible");
        } else {
            setCodeCellVisibility(inputField, "hidden");
        }
    }


    // Button constructor
    const hideCodeButton = id => `<input class="hidebtn" type="checkbox" id="hidebtn${id}" data-id="${id}"><label title="Toggle cell" for="hidebtn${id}" class="plusminus"><span class="pm_h"></span><span class="pm_v"></span></label>`

    var addHideButton = function () {
        // If a hide button is already added, don't add another
        if (document.querySelector('div.hidecode input') !== null) {
            return;
        }

        // Find the input cells and add a hide button
        document.querySelectorAll('div.input_area').forEach(function (item, index) {
            if (!item.classList.contains("hidecode")) {
                // Skip the cell if it doesn't have a hidecode class
                return;
            }

            const id = codeCellId(index)
            item.setAttribute('id', id);
            // Insert the button just inside the end of the next div
            item.querySelector('div').insertAdjacentHTML('beforeend', hideCodeButton(id))

            // Set up the visibility toggle
            hideLink = document.querySelector(`#${id} div.highlight + input + label`);
            hideLink.addEventListener('click', toggleCodeCellVisibility)
        });
    }


    // Initialize the hide buttos
    var initHiddenCells = function () {
        // Add hide buttons to the cells
        addHideButton();

        // Toggle the code cells that should be hidden
        document.querySelectorAll('div.hidecode input').forEach(function (item) {
            setCodeCellVisibility(item, 'hidden');
            item.checked = true;
        })
    }

    initFunction(initHiddenCells);

</script>

  <!-- Load custom website scripts -->
  <script src="/lab-wiki/assets/js/scripts.js" async></script>

  <!-- Load custom user CSS and JS  -->
  <script src="/lab-wiki/assets/custom/custom.js" async></script>
  <link rel="stylesheet" href="/lab-wiki/assets/custom/custom.css">

  <!-- Update interact links w/ REST param, is defined in includes so we can use templates -->
  

  <!-- Lunr search code - will only be executed on the /search page -->
  <script src="/lab-wiki/assets/js/lunr/lunr.min.js" type="text/javascript"></script>
  <script>var initQuery = function() {
  // See if we have a search box
  var searchInput = document.querySelector('input#lunr_search');
  if (searchInput === null) {
    return;
  }

  // Function to parse our lunr cache
  var idx = lunr(function () {
    this.field('title')
    this.field('excerpt')
    this.field('categories')
    this.field('tags')
    this.ref('id')

    this.pipeline.remove(lunr.trimmer)

    for (var item in store) {
      this.add({
        title: store[item].title,
        excerpt: store[item].excerpt,
        categories: store[item].categories,
        tags: store[item].tags,
        id: item
      })
    }
  });

  // Run search upon keyup
  searchInput.addEventListener('keyup', function () {
    var resultdiv = document.querySelector('#results');
    var query = document.querySelector("input#lunr_search").value.toLowerCase();
    var result =
      idx.query(function (q) {
        query.split(lunr.tokenizer.separator).forEach(function (term) {
          q.term(term, { boost: 100 })
          if(query.lastIndexOf(" ") != query.length-1){
            q.term(term, {  usePipeline: false, wildcard: lunr.Query.wildcard.TRAILING, boost: 10 })
          }
          if (term != ""){
            q.term(term, {  usePipeline: false, editDistance: 1, boost: 1 })
          }
        })
      });

      // Empty the results div
      while (resultdiv.firstChild) {
        resultdiv.removeChild(resultdiv.firstChild);
      }

    resultdiv.insertAdjacentHTML('afterbegin', '<p class="results__found">'+result.length+' Result(s) found</p>');
    for (var item in result) {
      var ref = result[item].ref;
      if(store[ref].teaser){
        var searchitem =
          '<div class="list__item">'+
            '<article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">'+
              '<h2 class="archive__item-title" itemprop="headline">'+
                '<a href="'+store[ref].url+'" rel="permalink">'+store[ref].title+'</a>'+
              '</h2>'+
              '<div class="archive__item-teaser">'+
                '<img src="'+store[ref].teaser+'" alt="">'+
              '</div>'+
              '<p class="archive__item-excerpt" itemprop="description">'+store[ref].excerpt.split(" ").splice(0,20).join(" ")+'...</p>'+
            '</article>'+
          '</div>';
      }
      else{
    	  var searchitem =
          '<div class="list__item">'+
            '<article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">'+
              '<h2 class="archive__item-title" itemprop="headline">'+
                '<a href="'+store[ref].url+'" rel="permalink">'+store[ref].title+'</a>'+
              '</h2>'+
              '<p class="archive__item-excerpt" itemprop="description">'+store[ref].excerpt.split(" ").splice(0,20).join(" ")+'...</p>'+
            '</article>'+
          '</div>';
      }
      resultdiv.insertAdjacentHTML('beforeend', searchitem);
    }
  });
};

initFunction(initQuery);
</script>
</head>

  <body>
    <!-- .js-show-sidebar shows sidebar by default -->
    <div id="js-textbook" class="c-textbook js-show-sidebar">
      



<nav id="js-sidebar" class="c-textbook__sidebar">
  <a href="https://xinhe-lab.github.io/lab-wiki"><img src="/lab-wiki/images/logo.png" class="textbook_logo" id="sidebar-logo" data-turbolinks-permanent/></a>
  <h2 class="c-sidebar__title">Lab Docs</h2>
  <ul class="c-sidebar__chapters">
    
      
      
        <li class="c-sidebar__chapter"><a class="c-sidebar__entry" href="/lab-wiki/search.html"><img src="/lab-wiki/images/search.svg" alt="Search The Site"></a></li>
        
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/welcome.html"
        >
          
          Home
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/orientation/index.html"
        >
          
          Orientation
        </a>

        

          
          
          
          

          

          <ul class="c-sidebar__sections ">
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/orientation/5m-git.html"
                >
                  
                  5 minutes git
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/orientation/github-issues.html"
                >
                  
                  Use github issues
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/orientation/debian-setup.html"
                >
                  
                  Debian based Linux setup
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/orientation/jupyter-setup.html"
                >
                  
                  Jupyter and SoS suite setup
                </a>

                
                

              </li>
              
            
          </ul>
        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/project_resource/index.html"
        >
          
          Projects
        </a>

        

          
          
          
          

          

          <ul class="c-sidebar__sections ">
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/project_resource/project.html"
                >
                  
                  Project repos
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/project_resource/investigation.html"
                >
                  
                  Prelim. explorations
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/project_resource/pipeline.html"
                >
                  
                  Pipelines & workflows
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/project_resource/data.html"
                >
                  
                  Data resources
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/project_resource/sharing-guideline.html"
                >
                  
                  Sharing guidelines
                </a>

                
                

              </li>
              
            
          </ul>
        
      </li>

      
    
      
      
        <li class="c-sidebar__divider"></li>
        
      
      
        <li><h2 class="c-sidebar__title">Tutorials</li>
        
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/computing_tutorial/index.html"
        >
          
          Computing
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/compbio_tutorial/index.html"
        >
          
          Computational biology
        </a>

        
      </li>

      
    
      
      
        <li class="c-sidebar__divider"></li>
        
      
      
        <li><h2 class="c-sidebar__title">Resources</li>
        
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/learning_center/index.html"
        >
          
          Learning resource
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/productivity_tips/index.html"
        >
          
          Productivity tips
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/this_that/index.html"
        >
          
          This and that
        </a>

        
      </li>

      
    
  </ul>
  <p class="sidebar_footer">Copyright (c) 2019 <br> <a href="http://xinhelab.org">Xin He Lab</a>.<br>Distributed under <a href="https://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA license</a>.<br></p>
</nav>

      
      <!-- Empty sidebar placeholder that we'll auto-fill with javascript -->
      <aside class="sidebar__right">
          <header><h4 class="nav__title"><i class="fa fa-list"></i>   Contents</h4></header>
          <nav class="onthispage">
          </nav>
      </aside>
      
      <main class="c-textbook__page" tabindex="-1">
          <div class="o-wrapper">
            <div class="c-sidebar-toggle">
  <!-- We show the sidebar by default so we use .is-active -->
  <button
    id="js-sidebar-toggle"
    class="hamburger hamburger--arrowalt is-active"
  >
    <span class="hamburger-box">
      <span class="hamburger-inner"></span>
    </span>
    <span class="c-sidebar-toggle__label">Toggle Sidebar</span>
  </button>
</div>

            

            <div class="c-textbook__content">
              <h2 id="download-data-from-dbgap">Download data from dbGaP</h2>

<p>A brief documentation to how dbGaP website is accessed &amp; how to download data.</p>

<h3 id="contact">Contact</h3>

<p>Gao Wang, Dongyue Xie</p>

<h3 id="data-directory">Data directory</h3>

<p>Files downloaded from dbGaP need to be decypted &amp; extracted to usable formats. Files are first to be downloaded to some temporary directory, then processed and saved into a more permanent location. It is therefore suggested that on RCC Midway, for example, you download data to a temporary folder under <code class="language-plaintext highlighter-rouge">/scratch/midway2/$USER</code> then extract them to shared computational space <code class="language-plaintext highlighter-rouge">/project2/mstephens/</code> for sharing with the group. This documentation assumes this workflow.</p>

<h3 id="dbgap-access">dbGaP access</h3>

<ol>
  <li>
    <p>Visit <a href="https://www.ncbi.nlm.nih.gov/gap">dbGaP</a>, click on <code class="language-plaintext highlighter-rouge">Controlled Access Data</code>, then login with your eRA Commons account. You’ll see the dataset previously applied for and approved under the tab <code class="language-plaintext highlighter-rouge">Authorized Access -&gt; My Requests</code>.</p>
  </li>
  <li>
    <p>In the column <code class="language-plaintext highlighter-rouge">Actions</code> click <code class="language-plaintext highlighter-rouge">Request Files</code> for the dataset you want to download. You should see two possibilites in two tabs:</p>
    <ul>
      <li>“Phenotype and Genotype files”</li>
      <li>“SRA data (reads and reference alignments)”</li>
    </ul>
  </li>
</ol>

<h3 id="download-phenotype-and-genotype-files">Download “Phenotype and Genotype files”</h3>

<p>Click on the tab <code class="language-plaintext highlighter-rouge">Phenotype and Genotype files</code>, then click / select data file names. After files you need are selected, create the data download request. You should be directed to a download instruction page. You should also receive an email confirmation which contains a link to this page.</p>

<p>For example, to download V8 eQTL release, use the Request for <code class="language-plaintext highlighter-rouge">Exchange Area</code> data then check the <code class="language-plaintext highlighter-rouge">+</code> sign to unfold directory and select data to download, in my case:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>gtex/exchange/GTEx_phs000424/exchange/analysis_releases/GTEx_Analysis_2017-06-05_v8/eqtl
</code></pre></div></div>

<p>The download instruction page offers several options but I find it most straightforward to use Linux command tool <code class="language-plaintext highlighter-rouge">ascp</code>.</p>

<p><code class="language-plaintext highlighter-rouge">ascp</code> can be download <a href="https://downloads.asperasoft.com/en/downloads/50">for Linux</a> then install using <code class="language-plaintext highlighter-rouge">bash</code> command. It requires root to install which we don’t have on RCC Midway. But it is easy to hack the installer: just open it with a text editor, eg <code class="language-plaintext highlighter-rouge">vim</code> and edit the first 10 lines of code to remove root requirement and change installation path, then save and run <code class="language-plaintext highlighter-rouge">bash</code> to install. You should find two files under a <code class="language-plaintext highlighter-rouge">bin</code> folder in the path you just edited to the installer: 1) the executable <code class="language-plaintext highlighter-rouge">ascp</code> and 2) the license file <code class="language-plaintext highlighter-rouge">aspera-license</code>. If you don’t know how to hack it, you can use a version I got on Midway <code class="language-plaintext highlighter-rouge">/project2/mstephens/software/ascp</code> (the license file sits in the same directory as <code class="language-plaintext highlighter-rouge">ascp</code>). Additionally you need an ssh key which <a href="https://www.ibm.com/support/pages/downloading-data-ncbi-command-line#locations">comes from the installation</a> but again I provide it at <code class="language-plaintext highlighter-rouge">/project2/mstephens/software/asperaweb_id_dsa.openssh</code>. Having <code class="language-plaintext highlighter-rouge">ascp</code> executable alone without these additional files is not going to work.</p>

<p>Now you can run the download command. On the instruction page the command might look like:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>"%ASPERA_CONNECT_DIR%\bin\ascp" -QTr -l 300M -k 1 -i "%ASPERA_CONNECT_DIR%\etc\asperaweb_id_dsa.openssh" -W A7A8C74EB00A14C92826A8EC64785904771847404B340486B977DD32F5F3701B399760E2328AA4F43ADB2A6E66A2EFA22D dbtest@gap-upload.ncbi.nlm.nih.gov:data/instant/gaowang/72550 .
</code></pre></div></div>

<p>but in our case you with <code class="language-plaintext highlighter-rouge">ascp</code> set up as previously discussed, you should change it to:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>/project2/mstephens/software/ascp -QTr -l 300M -k 1 -i /project2/mstephens/software/asperaweb_id_dsa.openssh -W A7A8C74EB00A14C92826A8EC64785904771847404B340486B977DD32F5F3701B399760E2328AA4F43ADB2A6E66A2EFA22D dbtest@gap-upload.ncbi.nlm.nih.gov:data/instant/gaowang/72550 /scratch/midway2/dyxie/dbGaP/
</code></pre></div></div>

<p>where in this example <code class="language-plaintext highlighter-rouge">/scratch/midway2/dyxie/dbGaP/</code> is your previous setup for download directory. Then you will find your downloaded file under a folder <code class="language-plaintext highlighter-rouge">72550</code> under that directory, as implied in commands above.</p>

<p><strong>Note: we need absolute path to <code class="language-plaintext highlighter-rouge">asperaweb_id_dsa.openssh</code> not relative path such as <code class="language-plaintext highlighter-rouge">./</code> – that won’t work!</strong> Ask Kushal he learned the lesson the hard way.</p>

<p>The downloaded data have <code class="language-plaintext highlighter-rouge">.ncbi_enc</code> file extensions. We need <code class="language-plaintext highlighter-rouge">sra-toolkit</code> to decode the data next.</p>

<h3 id="install-sra-toolkit">Install <code class="language-plaintext highlighter-rouge">sra-toolkit</code></h3>

<p>If use Anaconda on midway2, then it does not allow user to install <code class="language-plaintext highlighter-rouge">sra-toolkit</code>. So we need to install the conda by ourthelves:</p>

<p>First download and install <code class="language-plaintext highlighter-rouge">Miniconda3</code> if you haven’t done that before. Replace <code class="language-plaintext highlighter-rouge">%Download address%</code> to the download link that <a href="https://docs.conda.io/en/latest/miniconda.html">you can find here</a>:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>wget %Download address%

</code></pre></div></div>
<p>Then run</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>bash Miniconda3-latest-Linux-x86_64.sh -bfp /scratch/midway2/$USER/miniconda3
echo "export PATH=/scratch/midway2/$USER/miniconda3/bin:\$PATH" &gt;&gt; ~/.bashrc
source ~/.bashrc
</code></pre></div></div>

<p>Check if using the right conda</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>which conda
</code></pre></div></div>

<p>Then the sra tool can be installed via:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>conda install -c bioconda sra-tools 
</code></pre></div></div>

<h3 id="set-dbgap-repository-key-for-the-project">Set dbGaP repository key for the project</h3>

<p>In order to decrypt, a dbGaP repository key is also required. The key is provided in a file with suffix “.ngc”. It can be obtained from two places in PI’s dbGaP account.</p>

<ol>
  <li>The first place is the project page under “My Projects” tab, through a link named “get dbGaP repository key” in the “Actions” column. The key downloaded from here is valid to all downloaded data under the project.</li>
  <li>The second place is the download page under “Downloads” tab, through a link named “get dbGaP repository key in the “Actions” column.</li>
</ol>

<p>I get the key from the first place. Then config the key by running command</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>vdb-config --import ~/prj_3253_D17102.ngc
</code></pre></div></div>

<p>After configuration, manually change the <code class="language-plaintext highlighter-rouge">mkfg</code> file:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>vim ~/.ncbi/user-settings.mkfg 
</code></pre></div></div>

<p>Then change the root to the data folder you store the data. For example, here, the download directory is <code class="language-plaintext highlighter-rouge">/scratch/midway2/dyxie/dbGaP/</code>. See below the change I made to point to that directory.</p>

<div class="language-diff highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">15c17
</span><span class="gd">&lt; /repository/user/protected/dbGaP-3253/root = "/home/dyxie/ncbi/dbGaP-3253"
</span><span class="p">---
</span><span class="gi">&gt; /repository/user/protected/dbGaP-3253/root = "/scratch/midway2/dyxie/dbGaP/"
</span></code></pre></div></div>

<p>Finally, run</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cd</span> /path/to/downloaded/data <span class="c"># in the above setting it is: /scratch/midway2/dyxie/dbGaP/72250</span>
vdb-decrypt <span class="nt">--decrypt-sra-files</span> <span class="nb">.</span> /path/to/destination <span class="nt">-v</span> <span class="c"># we can use somewhere under /project2/mstephens for the otput destination directory</span>
</code></pre></div></div>

<p>Then all the <code class="language-plaintext highlighter-rouge">*.ncbi_enc</code> files under <code class="language-plaintext highlighter-rouge">/path/to/downloaded/data</code> will be decrypted and the original <code class="language-plaintext highlighter-rouge">*.ncbi_enc</code> files will be automatically removed.</p>

<h3 id="download-sra-data-reads-and-reference-alignments">Download “SRA data (reads and reference alignments)”</h3>

<p>Before this, please follow instructions above in sections “Install <code class="language-plaintext highlighter-rouge">sra-toolkit</code>” and “Set dbGaP repository key for the project”.</p>

<p>Then, click ‘SRA RUN selector’, then check ‘data-type(Run)’ from the Filters List on the left side. Then check <code class="language-plaintext highlighter-rouge">rna</code>.</p>

<p>Select all the sra files you want to download and click ‘Selected’ in the Select session. Now download the <code class="language-plaintext highlighter-rouge">Accession List</code>. The downloaded list is named <code class="language-plaintext highlighter-rouge">SRR_Acc_List.txt</code>, in which are a list of SRR IDs.</p>

<p>To download these files and convert to BAM files, run the folloiwng command, replace <code class="language-plaintext highlighter-rouge">xyz</code> with the ID.</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /path/to/download/dir
prefetch xyz
sam-dump xyz.sra | samtools view -Sb &gt; /path/to/destination/xyz.bam
</code></pre></div></div>

              <nav class="c-page__nav">
  

  
</nav>

            </div>
          </div>
        </div>
      </main>
    </div>

  </body>
</html>
