<!DOCTYPE html>
<html lang="en">
  

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width,minimum-scale=1">

  <title>comp-genomics-2016</title>
  <meta name="description" content="">

  <link rel="canonical" href="https://xinhe-lab.github.io/lab-wiki/learning_center/comp-genomics-2016.html">
  <link rel="alternate" type="application/rss+xml" title="Lab Docs" href="https://xinhe-lab.github.io/lab-wiki/feed.xml">

  <meta property="og:url"         content="https://xinhe-lab.github.io/lab-wiki/learning_center/comp-genomics-2016.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="comp-genomics-2016" />
<meta property="og:description" content="" />
<meta property="og:image"       content="" />


  <script type="application/ld+json">
  {
  "@context": "http://schema.org",
  "@type": "NewsArticle",
  "mainEntityOfPage":
    "https://xinhe-lab.github.io/lab-wiki/learning_center/comp-genomics-2016.html",
  "headline":
    "comp-genomics-2016",
  "datePublished":
    "2019-12-17T15:37:14-06:00",
  "dateModified":
    "2019-12-17T15:37:14-06:00",
  "description":
    "",
  "author": {
    "@type": "Person",
    "name": "Gao Wang"
  },
  "publisher": {
    "@type": "Organization",
    "name": "Data 100 at UC Berkeley",
    "logo": {
      "@type": "ImageObject",
      "url": "https://xinhe-lab.github.io/lab-wiki",
      "width": 60,
      "height": 60
    }
  },
  "image": {
    "@type": "ImageObject",
    "url": "https://xinhe-lab.github.io/lab-wiki",
    "height": 60,
    "width": 60
  }
}

  </script>
  <link rel="stylesheet" href="/lab-wiki/assets/css/styles.css">
  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css ">
  <link rel="apple-touch-icon" sizes="57x57" href="/apple-touch-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/apple-touch-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/apple-touch-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/apple-touch-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/apple-touch-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/apple-touch-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/apple-touch-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/apple-touch-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon-180x180.png">

  <!-- <link rel="manifest" href="/manifest.json"> -->
  <!-- <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#efae0a"> -->
  <meta name="msapplication-TileColor" content="#da532c">
  <meta name="msapplication-TileImage" content="/mstile-144x144.png">
  <meta name="theme-color" content="#233947">

  <!-- Favicon -->
  <link rel="shortcut icon" type="image/x-icon" href="/lab-wiki/images/favicon.ico">

  <!-- MathJax Config -->
  <!-- Allow inline math using $ and automatically break long math lines -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
    tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true,
        processEnvironments: true
    },
    CommonHTML: {
        linebreaks: {
            automatic: true,
        },
    },
});
</script>
<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS_CHTML' async></script>

  <!-- DOM updating function -->
  <script>
const runWhenDOMLoaded = cb => {
  if (document.readyState != 'loading') {
    cb()
  } else if (document.addEventListener) {
    document.addEventListener('DOMContentLoaded', cb)
  } else {
    document.attachEvent('onreadystatechange', function() {
      if (document.readyState == 'complete') cb()
    })
  }
}

// Helper function to init things quickly
initFunction = function(myfunc) {
  runWhenDOMLoaded(myfunc);
  document.addEventListener('turbolinks:load', myfunc);
};
</script>

  <!-- Define some javascript variables that will be useful in other javascript -->
  <script>
    const site_basename = '/lab-wiki';
  </script>

  <!-- Add AnchorJS to let headers be linked -->
  <script src="/lab-wiki/assets/js/anchor.min.js"  type="text/javascript"></script>
  <script>

initFunction(function () {
    anchors.add("main h1, main h2, main h3, main h4")
});

</script>

  <!-- Include Turbolinks to make page loads fast -->
  <!-- https://github.com/turbolinks/turbolinks -->
  <script src="/lab-wiki/assets/js/turbolinks.js" async></script>
  <meta name="turbolinks-cache-control" content="no-cache">

  <!-- Load nbinteract for widgets -->
  

  <!-- Load Thebelab for interactive widgets -->
  <!-- Include Thebelab for interactive code if it's enabled -->



  <!-- Load the auto-generating TOC -->
  <script src="/lab-wiki/assets/js/tocbot.min.js"  type="text/javascript"></script>
  <script>
var initToc = function () {
  tocbot.init({
    tocSelector: 'nav.onthispage',
    contentSelector: '.c-textbook__content',
    headingSelector: 'h2, h3',
    orderedList: false,
    collapseDepth: 6,
    listClass: 'toc__menu',
    activeListItemClass: "",  // Not using
    activeLinkClass: "", // Not using
  });
  tocbot.refresh();
}
initFunction(initToc);
</script>

  <!-- Google analytics -->
  <script src="/lab-wiki/assets/js/ga.js" async></script>

  <!-- Clipboard copy button -->
  <script src="https://cdn.jsdelivr.net/npm/clipboard@2/dist/clipboard.min.js" async></script>

  <!-- Load JS that depends on site variables -->
  <script>
/**
 * Set up copy/paste for code blocks
 */
const codeCellId = index => `codecell${index}`

const clipboardButton = id =>
  `<a id="copy-button-${id}" class="btn copybtn o-tooltip--left" data-tooltip="Copy" data-clipboard-target="#${id}">
    <img src="/lab-wiki/assets/images/copy-button.svg" alt="Copy to clipboard">
  </a>`

// Clears selected text since ClipboardJS will select the text when copying
const clearSelection = () => {
  if (window.getSelection) {
    window.getSelection().removeAllRanges()
  } else if (document.selection) {
    document.selection.empty()
  }
}

// Changes tooltip text for two seconds, then changes it back
const temporarilyChangeTooltip = (el, newText) => {
  const oldText = el.getAttribute('data-tooltip')
  el.setAttribute('data-tooltip', newText)
  setTimeout(() => el.setAttribute('data-tooltip', oldText), 2000)
}

const addCopyButtonToCodeCells = () => {
  // If ClipboardJS hasn't loaded, wait a bit and try again. This
  // happens because we load ClipboardJS asynchronously.
  if (window.ClipboardJS === undefined) {
    setTimeout(addCopyButtonToCodeCells, 250)
    return
  }

  const codeCells = document.querySelectorAll('div.c-textbook__content > div.highlighter-rouge > div.highlight > pre, div.input_area pre')
  codeCells.forEach((codeCell, index) => {
    const id = codeCellId(index)
    codeCell.setAttribute('id', id)
    if (document.getElementById("copy-button" + id) == null) {
      codeCell.insertAdjacentHTML('afterend', clipboardButton(id));
    }
  })

  const clipboard = new ClipboardJS('.copybtn')
  clipboard.on('success', event => {
    clearSelection()
    temporarilyChangeTooltip(event.trigger, 'Copied!')
  })

  clipboard.on('error', event => {
    temporarilyChangeTooltip(event.trigger, 'Failed to copy')
  })

  // Get rid of clipboard before the next page visit to avoid memory leak
  document.addEventListener('turbolinks:before-visit', () =>
    clipboard.destroy()
  )
}

initFunction(addCopyButtonToCodeCells);
</script>


  <!-- Hide cell code -->
  <script>
    /**
    Add buttons to hide code cells
    */


    var setCodeCellVisibility = function (inputField, kind) {
        // Update the image and class for hidden
        var id = inputField.getAttribute('data-id');
        var codeCell = document.querySelector(`#${id} div.highlight`);

        if (kind === "visible") {
            codeCell.classList.remove('hidden');
            inputField.checked = true;
        } else {
            codeCell.classList.add('hidden');
            inputField.checked = false;
        }
    }

    var toggleCodeCellVisibility = function (event) {
        // The label is clicked, and now we decide what to do based on the input field's clicked status
        if (event.target.tagName === "LABEL") {
            var inputField = event.target.previousElementSibling;
        } else {
            // It is the span inside the target
            var inputField = event.target.parentElement.previousElementSibling;
        }

        if (inputField.checked === true) {
            setCodeCellVisibility(inputField, "visible");
        } else {
            setCodeCellVisibility(inputField, "hidden");
        }
    }


    // Button constructor
    const hideCodeButton = id => `<input class="hidebtn" type="checkbox" id="hidebtn${id}" data-id="${id}"><label title="Toggle cell" for="hidebtn${id}" class="plusminus"><span class="pm_h"></span><span class="pm_v"></span></label>`

    var addHideButton = function () {
        // If a hide button is already added, don't add another
        if (document.querySelector('div.hidecode input') !== null) {
            return;
        }

        // Find the input cells and add a hide button
        document.querySelectorAll('div.input_area').forEach(function (item, index) {
            if (!item.classList.contains("hidecode")) {
                // Skip the cell if it doesn't have a hidecode class
                return;
            }

            const id = codeCellId(index)
            item.setAttribute('id', id);
            // Insert the button just inside the end of the next div
            item.querySelector('div').insertAdjacentHTML('beforeend', hideCodeButton(id))

            // Set up the visibility toggle
            hideLink = document.querySelector(`#${id} div.highlight + input + label`);
            hideLink.addEventListener('click', toggleCodeCellVisibility)
        });
    }


    // Initialize the hide buttos
    var initHiddenCells = function () {
        // Add hide buttons to the cells
        addHideButton();

        // Toggle the code cells that should be hidden
        document.querySelectorAll('div.hidecode input').forEach(function (item) {
            setCodeCellVisibility(item, 'hidden');
            item.checked = true;
        })
    }

    initFunction(initHiddenCells);

</script>

  <!-- Load custom website scripts -->
  <script src="/lab-wiki/assets/js/scripts.js" async></script>

  <!-- Load custom user CSS and JS  -->
  <script src="/lab-wiki/assets/custom/custom.js" async></script>
  <link rel="stylesheet" href="/lab-wiki/assets/custom/custom.css">

  <!-- Update interact links w/ REST param, is defined in includes so we can use templates -->
  

  <!-- Lunr search code - will only be executed on the /search page -->
  <script src="/lab-wiki/assets/js/lunr/lunr.min.js" type="text/javascript"></script>
  <script>var initQuery = function() {
  // See if we have a search box
  var searchInput = document.querySelector('input#lunr_search');
  if (searchInput === null) {
    return;
  }

  // Function to parse our lunr cache
  var idx = lunr(function () {
    this.field('title')
    this.field('excerpt')
    this.field('categories')
    this.field('tags')
    this.ref('id')

    this.pipeline.remove(lunr.trimmer)

    for (var item in store) {
      this.add({
        title: store[item].title,
        excerpt: store[item].excerpt,
        categories: store[item].categories,
        tags: store[item].tags,
        id: item
      })
    }
  });

  // Run search upon keyup
  searchInput.addEventListener('keyup', function () {
    var resultdiv = document.querySelector('#results');
    var query = document.querySelector("input#lunr_search").value.toLowerCase();
    var result =
      idx.query(function (q) {
        query.split(lunr.tokenizer.separator).forEach(function (term) {
          q.term(term, { boost: 100 })
          if(query.lastIndexOf(" ") != query.length-1){
            q.term(term, {  usePipeline: false, wildcard: lunr.Query.wildcard.TRAILING, boost: 10 })
          }
          if (term != ""){
            q.term(term, {  usePipeline: false, editDistance: 1, boost: 1 })
          }
        })
      });

      // Empty the results div
      while (resultdiv.firstChild) {
        resultdiv.removeChild(resultdiv.firstChild);
      }

    resultdiv.insertAdjacentHTML('afterbegin', '<p class="results__found">'+result.length+' Result(s) found</p>');
    for (var item in result) {
      var ref = result[item].ref;
      if(store[ref].teaser){
        var searchitem =
          '<div class="list__item">'+
            '<article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">'+
              '<h2 class="archive__item-title" itemprop="headline">'+
                '<a href="'+store[ref].url+'" rel="permalink">'+store[ref].title+'</a>'+
              '</h2>'+
              '<div class="archive__item-teaser">'+
                '<img src="'+store[ref].teaser+'" alt="">'+
              '</div>'+
              '<p class="archive__item-excerpt" itemprop="description">'+store[ref].excerpt.split(" ").splice(0,20).join(" ")+'...</p>'+
            '</article>'+
          '</div>';
      }
      else{
    	  var searchitem =
          '<div class="list__item">'+
            '<article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">'+
              '<h2 class="archive__item-title" itemprop="headline">'+
                '<a href="'+store[ref].url+'" rel="permalink">'+store[ref].title+'</a>'+
              '</h2>'+
              '<p class="archive__item-excerpt" itemprop="description">'+store[ref].excerpt.split(" ").splice(0,20).join(" ")+'...</p>'+
            '</article>'+
          '</div>';
      }
      resultdiv.insertAdjacentHTML('beforeend', searchitem);
    }
  });
};

initFunction(initQuery);
</script>
</head>

  <body>
    <!-- .js-show-sidebar shows sidebar by default -->
    <div id="js-textbook" class="c-textbook js-show-sidebar">
      



<nav id="js-sidebar" class="c-textbook__sidebar">
  <a href="https://xinhe-lab.github.io/lab-wiki"><img src="/lab-wiki/images/logo.png" class="textbook_logo" id="sidebar-logo" data-turbolinks-permanent/></a>
  <h2 class="c-sidebar__title">Lab Docs</h2>
  <ul class="c-sidebar__chapters">
    
      
      
        <li class="c-sidebar__chapter"><a class="c-sidebar__entry" href="/lab-wiki/search.html"><img src="/lab-wiki/images/search.svg" alt="Search The Site"></a></li>
        
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/welcome.html"
        >
          
          Home
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/orientation/index.html"
        >
          
          Orientation
        </a>

        

          
          
          
          

          

          <ul class="c-sidebar__sections ">
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/orientation/5m-git.html"
                >
                  
                  5 minutes git
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/orientation/github-issues.html"
                >
                  
                  Use github issues
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/orientation/debian-setup.html"
                >
                  
                  Debian based Linux setup
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/orientation/jupyter-setup.html"
                >
                  
                  Jupyter and SoS suite setup
                </a>

                
                

              </li>
              
            
          </ul>
        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/project_resource/index.html"
        >
          
          Projects
        </a>

        

          
          
          
          

          

          <ul class="c-sidebar__sections ">
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/project_resource/project.html"
                >
                  
                  Project repositories
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/project_resource/pipeline.html"
                >
                  
                  Pipelines & workflows
                </a>

                
                

              </li>
              
            
          </ul>
        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/shared_computing/index.html"
        >
          
          Shared computing
        </a>

        

          
          
          
          

          

          <ul class="c-sidebar__sections ">
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/shared_computing/data.html"
                >
                  
                  Data resources
                </a>

                
                

              </li>
              
            
              
              

              <li class="c-sidebar__section">
                <a class="c-sidebar__entry "
                  href="/lab-wiki/shared_computing/sharing-guideline.html"
                >
                  
                  Sharing guidelines
                </a>

                
                

              </li>
              
            
          </ul>
        
      </li>

      
    
      
      
        <li class="c-sidebar__divider"></li>
        
      
      
        <li><h2 class="c-sidebar__title">Tutorials</li>
        
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/computing_tutorial/index.html"
        >
          
          Computing
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/compbio_tutorial/index.html"
        >
          
          Computational biology
        </a>

        
      </li>

      
    
      
      
        <li class="c-sidebar__divider"></li>
        
      
      
        <li><h2 class="c-sidebar__title">Resources</li>
        
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/learning_center/index.html"
        >
          
          Learning resource
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/research_resource/index.html"
        >
          
          Research resource
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/productivity_tips/index.html"
        >
          
          Productivity tips
        </a>

        
      </li>

      
    
      
      

      
      

      
      
      <li class="c-sidebar__chapter">
        <a class="c-sidebar__entry "
          href="/lab-wiki/this_that/index.html"
        >
          
          This and that
        </a>

        
      </li>

      
    
  </ul>
  <p class="sidebar_footer">Copyright (c) 2019 <br> <a href="http://xinhelab.org">Xin He Lab</a>.<br>Distributed under <a href="https://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA license</a>.<br></p>
</nav>

      
      <!-- Empty sidebar placeholder that we'll auto-fill with javascript -->
      <aside class="sidebar__right">
          <header><h4 class="nav__title"><i class="fa fa-list"></i>   Contents</h4></header>
          <nav class="onthispage">
          </nav>
      </aside>
      
      <main class="c-textbook__page" tabindex="-1">
          <div class="o-wrapper">
            <div class="c-sidebar-toggle">
  <!-- We show the sidebar by default so we use .is-active -->
  <button
    id="js-sidebar-toggle"
    class="hamburger hamburger--arrowalt is-active"
  >
    <span class="hamburger-box">
      <span class="hamburger-inner"></span>
    </span>
    <span class="c-sidebar-toggle__label">Toggle Sidebar</span>
  </button>
</div>

            

            <div class="c-textbook__content">
              <h1 id="computational-genomics-reading-list-version-2016">Computational genomics reading list (version 2016)</h1>

<p>A list compiled by Xin He at UChicago in 2016.</p>

<h2 id="sequence-analysis">Sequence Analysis</h2>

<h3 id="motif-discovery">Motif discovery</h3>

<h4 id="background">Background</h4>

<ul>
  <li>What are DNA sequence motifs? NBT, 2006</li>
  <li>How does DNA sequence motif discovery work? NBT, 2006</li>
</ul>

<h4 id="paper">Paper</h4>

<ul>
  <li>Fitting a mixture model by expectation maximization to discover motifs in biopolymers, ISMB, 2004 (MEME)</li>
</ul>

<h3 id="profile-hmm">Profile HMM</h3>
<h4 id="background-1">Background</h4>

<ul>
  <li>Eddy, Hidden Markov Models, Current Opinion in Structural Biology, 1996</li>
  <li>DEKM, Introduction in Chapter 5.</li>
</ul>

<h4 id="paper-1">Paper</h4>

<p>DEKM Chapter 5, 5.1-5.4.</p>

<h3 id="dynamic-programming">Dynamic Programming</h3>

<ul>
  <li>Introduction to Algorithms, CLRS, 3^rd^ edition. Sections: 15.1-15.3.
    <ul>
      <li>Remark: focus on the main recurrence equations, (15.2) and (15.7). Pay less attention to technicalities such as running time analysis.</li>
    </ul>
  </li>
</ul>

<h3 id="sequence-alignment-and-hmm">Sequence alignment and HMM</h3>

<ul>
  <li>DEKM, Sections 2.1-2.3, up to “Local alignment”.</li>
  <li>DEKM, Sections 3.2-3.3 (ignoring the last part, “Modeling of labeled sequences”), 11.6.</li>
</ul>

<h2 id="ngs-data-analysis">NGS Data Analysis</h2>

<h3 id="variant-calling">Variant calling</h3>

<h4 id="background-2">Background</h4>

<ul>
  <li>Genotype and SNP calling from next-generation sequencing data [Nielsen &amp; Song, NRG, 2011]</li>
</ul>

<h4 id="paper-2">Paper</h4>

<ul>
  <li>A framework for variation discovery and genotyping using next-generation DNA sequencing data, NG, 2011 (GATK)</li>
</ul>

<h3 id="chip-seq">ChIP-seq</h3>

<h4 id="background-3">Background</h4>

<ul>
  <li>Computational methodology for ChIP-seq analysis, Quant Biology, 2013</li>
</ul>

<h4 id="paper-3">Paper</h4>

<ul>
  <li>Model-based Analysis of ChIP-Seq, Genome Bio, 2008 (MACS)</li>
</ul>

<h3 id="rna-seq-transcriptome-inference">RNA-seq: transcriptome inference</h3>

<h4 id="background-4">Background</h4>

<ul>
  <li>RNA-Seq: a revolutionary tool for transcriptomics, NRG, 2009</li>
</ul>

<h4 id="paper-4">Paper</h4>

<ul>
  <li>Statistical Inferences for Isoform Expression in RNA-Seq. Bioinformatics, 2009</li>
</ul>

<h3 id="rna-seq-differential-expression">RNA-seq: differential expression</h3>

<h4 id="background-5">Background</h4>

<h4 id="paper-5">Paper</h4>

<ul>
  <li>Small-sample estimation of negative binomial dispersion, with applications to SAGE data, Biostatistics, 2008 (edgeR)</li>
  <li>Moderated statistical tests for assessing differences in tag abundance, Bioinformatics, 2007 (edgeR)</li>
</ul>

<h2 id="regulatory-genomics">Regulatory Genomics</h2>

<h3 id="gene-regulatory-networks">Gene regulatory networks</h3>

<h4 id="background-6">Background</h4>

<ul>
  <li>Inferring Cellular Networks Using Probabilistic Graphical Models, Science, 2004</li>
</ul>

<h4 id="paper-6">Paper</h4>

<ul>
  <li>Using Bayesian Networks to Analyze Expression Data, J Computational Biology, 2000</li>
</ul>

<h3 id="from-sequence-to-gene-expression">From sequence to gene expression</h3>

<h4 id="background-7">Background</h4>
<h4 id="paper-7">Paper</h4>
<ul>
  <li>Integrating regulatory motif discovery and genome-wide expression analysis, PNAS, 2003</li>
  <li>Genome-wide discovery of transcriptional modules from DNA sequence and gene expression. Bioinformatics, 2003</li>
</ul>

<h3 id="epigenomics">Epigenomics</h3>

<h4 id="background-8">Background</h4>

<ul>
  <li>Mapping Human Epigenomes [Rivera &amp; Ren, Cell, 2013]</li>
</ul>

<h4 id="paper-8">Paper</h4>

<ul>
  <li>Discovery and characterization of chromatin states for systematic annotation of the human genome, NBT, 2010 (ChromHMM)</li>
</ul>

<h3 id="predicting-transcription-factor-binding">Predicting transcription factor binding</h3>

<h4 id="background-9">Background</h4>
<h4 id="paper-9">Paper</h4>

<ul>
  <li>Accurate inference of transcription factor binding from DNA sequence and chromatin accessibility data, Genome Research. 2011 (CENTIPEDE)</li>
</ul>

<h2 id="molecular-evolution">Molecular Evolution</h2>

<h3 id="molecular-evolution-and-comparative-genomics">Molecular evolution and comparative genomics</h3>

<h4 id="background-10">Background</h4>

<ul>
  <li>DEKM Chapter 8, 8.1-8.3</li>
</ul>

<h4 id="paper-10">Paper</h4>

<ul>
  <li>Evolutionarily conserved elements in vertebrate, insect, worm, and yeast genomes. Genome Res, 2005 (PhastCons)</li>
  <li>Phylogenetic hidden Markov models. In R. Nielsen, ed., Statistical Methods in Molecular Evolution, 2005. (Technical version of the PhastCons paper)</li>
</ul>

<h3 id="phylogeny-reconstruction">Phylogeny reconstruction</h3>

<h4 id="background-11">Background</h4>

<ul>
  <li>DEKM Chapter 8.4.</li>
  <li>Bayesian Inference of Phylogeny and Its Impact on Evolutionary Biology, Science, 2001</li>
</ul>

<h4 id="paper-11">Paper</h4>

<ul>
  <li>Markov chain monte carlo algorithms for the Bayesian analysis of phylogenetic trees, MBE, 1999</li>
</ul>

<h2 id="statistical-genetics">Statistical Genetics</h2>

<h3 id="methods-for-gwas">Methods for GWAS</h3>

<h4 id="background-12">Background</h4>

<ul>
  <li>Bayesian statistical methods for genetic association studies. NRG, 2009</li>
</ul>

<h4 id="paper-12">Paper</h4>

<ul>
  <li>Imputation-based analysis of association studies: candidate regions and quantitative traits. PLG, 2007</li>
</ul>

<h3 id="eqtl-mapping">eQTL mapping</h3>

<h4 id="background-13">Background</h4>

<h4 id="paper-13">Paper</h4>

<ul>
  <li>A statistical framework for joint eQTL analysis in multiple tissues. PLoS Genetics, 2013</li>
</ul>

<h3 id="systems-genetics">Systems genetics</h3>

<h4 id="background-14">Background</h4>

<h4 id="paper-14">Paper</h4>

<ul>
  <li>Sherlock: Detecting Gene-Disease Associations by Matching Patterns of Expression QTL and GWAS. AJHG, 2013</li>
</ul>

<h3 id="integrative-genetics-using-annotations">Integrative genetics using annotations</h3>

<h4 id="background-15">Background</h4>

<h4 id="paper-15">Paper</h4>

<ul>
  <li>Joint analysis of functional genomic data and genome-wide association studies of 18 human traits, The American Journal of Human Genetics. 2014</li>
</ul>

<h2 id="reference">Reference</h2>

<ul>
  <li>Biological Sequence Analysis, Durbin et al (DEKM)</li>
  <li>Introduction to algorithms (CLRS)</li>
</ul>

              <nav class="c-page__nav">
  

  
</nav>

            </div>
          </div>
        </div>
      </main>
    </div>

  </body>
</html>
